### PR TITLE
11582 Fix html validation errors PNI homepage

### PR DESCRIPTION
**Pull request by tbrlpld**
# Description

This PR fixes 15 HTML validation errors and 1 HTML validation warning on the PNI homepage. 

There are still 3 errors and 1 warning outstanding. The efforts required for those fixes seemed to large to include in this PR as they might necessitate full component refactors. Follow-up tickets have been filed (#11605, #11606 and #11607). 

Link to sample test page:
Related PRs/issues: #11582

# Checklist

**Tests**
- [x] Is the code I'm adding covered by tests?

**Changes in Models:**
- [-] Did I update or add new fake data?
- [-] Did I squash my migration?
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [-] Is my code documented?
- [-] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

### CODE REVIEW

File: network-api/networkapi/templates/fragments/buyersguide/pni_mobile_nav.html [hunk 1]
```diff
@@ -7,21 +7,17 @@
...
-                                            {{ current_category.name }}
-                                        {% else %}
-                                            {% trans "All Categories" %}
-                                        {% endif %}
-                                    </div>
-                                </div>
-                                <div class="d-inline-block align-self-center control"></div>
-                            </div>
+                        <button class="expander d-flex justify-content-between">
+                            <span class="active-link-label d-inline-block multipage-link active">
```
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/buyersguide/pni_mobile_nav.html:11]: Converted divs to spans

File: network-api/networkapi/templates/fragments/buyersguide/pni_mobile_nav.html [hunk 2]
```diff
@@ -7,21 +7,17 @@
             <div class="col-12 tw-border-0">
                 <div class="dropdown-nav">
                     <div>
-                        <button class="expander">
-                            <div class="d-flex justify-content-between">
-                                <div>
```
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/buyersguide/pni_mobile_nav.html:12]: Removed empty wrappers

File: network-api/networkapi/templates/fragments/buyersguide/pni_mobile_search.html [hunk 3]
```diff
@@ -7,11 +7,13 @@
                 <img
                     class="tw-w-8 tw-h-8 tw-mr-4"
                     src="{% static "_images/buyers-guide/pni-search.svg" %}"
+                    alt=""
                 >
                 <input
                     id="pni-mobile-bar"
                     type="text"
                     class="tw-w-full tw-bg-blue-05 tw-outline-none tw-py-7 tw-text-base tw-text-blue-60 placeholder:tw-text-blue-60"
+                    name="search"
```
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/buyersguide/pni_mobile_search.html:16]: For better announcement when the placeholder is overridden / a value is entered.

File: network-api/networkapi/templates/fragments/buyersguide/pni_sort_dropdown.html [hunk 4]
```diff
@@ -29,11 +36,11 @@
     </li>
 
 
-    <li aria-expanded="false" role="list" class="tw-absolute tw-top-0 tw-hidden tw-left-0 medium:tw-left-auto medium:tw-right-0 pni-creepiness__list-container tw-w-full medium:tw-w-max tw-z-50 tw-bg-white tw-border tw-border-gray-20 tw-py-2">
```
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/buyersguide/pni_sort_dropdown.html:32]: Invalid role

File: network-api/networkapi/templates/fragments/buyersguide/primary_nav.html [hunk 5]
```diff
@@ -46,7 +49,7 @@
                           tw-bg-no-repeat
                           tw-bg-center
                          "
-                   aria-label="Search input"
+                   aria-label="Search input toggle"
```
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/buyersguide/primary_nav.html:52]: So make the use a bit clearer.

File: network-api/networkapi/templates/pages/buyersguide/catalog.html [hunk 6]
```diff
@@ -2,6 +2,16 @@
 
 {% load static env i18n static wagtailimages_tags wagtailroutablepage_tags wagtailcore_tags cache bg_nav_tags localization %}
 
+{% block extended_head %}
+    <noscript>
+        <style>
```
Comment (tbrlpld) [inline network-api/networkapi/templates/pages/buyersguide/catalog.html:7]: `noscript` with nested `style` is only valid in the header.

File: network-api/networkapi/templates/pages/buyersguide/catalog.html [hunk 7]
```diff
@@ -79,7 +89,7 @@
                     <div class="tw-flex tw-mb-4 medium:tw-mb-0 tw-items-end tw-space-x-2 tw-overflow-auto tw-pb-2 tw-no-scrollbar tw-touch-pan-x tw-w-full subcategory-header medium:tw-mr-3 medium:tw-pr-5 medium:[mask:linear-gradient(to_right,black_calc(100%-24px),transparent)]">
 
                         <span id="product-filter-pni" class="tw-flex tw-cursor-pointer tw-text-gray-60 border tw-border-gray-20 tw-px-4 tw-py-2 tw-font-sans tw-rounded-3xl tw-font-normal tw-text-[12px] tw-leading-[1.3] tw-bg-white hover:tw-border-blue-10 hover:tw-bg-blue-10">
-                            <input type="checkbox" id="product-filter-pni-toggle" autocomplete="off">
```
Comment (tbrlpld) [inline network-api/networkapi/templates/pages/buyersguide/catalog.html:82]: Invalid attribute

File: source/images/buyers-guide/asterisk-mini.svg
Comment (tbrlpld) [inline source/images/buyers-guide/asterisk-mini.svg:1]: Fixed typo

**Review by danielfmiranda — CHANGES_REQUESTED** [review]
Hi @tbrlpld!

Thanks very much for taking this work on, after taking a look through the code changes, all updates make sense to me!

However, I did notice that on my local machine and on a [review app](https://foundation-s-11582-fix--lezajp.herokuapp.com/en/privacynotincluded/), it appears that the mobile dropdown navigation has stopped working.


If you could get it working again, I would be happy to approve this PR and the rest of the code changes! If I had to guess, it would be something related to the JS. If you need help or would like to pair-program through this, let me know and I would be happy to help where I can 👍

**Comment by tbrlpld** [comment]
> Hi @tbrlpld!
> 
> Thanks very much for taking this work on, after taking a look through the code changes, all updates make sense to me!
> 
> However, I did notice that on my local machine and on a [review app](https://foundation-s-11582-fix--lezajp.herokuapp.com/en/privacynotincluded/), it appears that the mobile dropdown navigation has stopped working.
> 
> If you could get it working again, I would be happy to approve this PR and the rest of the code changes! If I had to guess, it would be something related to the JS. If you need help or would like to pair-program through this, let me know and I would be happy to help where I can 👍

Thanks for the review @danielfmiranda. Yea, I noticed the issue with the category dropdown as well. However, this was already present in the `2023-eoy` branch before my changes. I have filed this [bug fix ticket](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/11604) to deal with that.

**Pull request merged**